website/addons/dryad/views.py

```diff
@@ -0,0 +1,225 @@
+from pyDryad import Dryad_DataOne, Dryad_SOLR
+
+import logging
+
+from flask import request
+from framework.flask import redirect
+from urllib2 import HTTPError
+
+from website.project.decorators import must_have_addon, must_be_valid_project
+from website.project.views.node import _view_project
+
+from website.util import rubeus
+
+from website.project.decorators import must_be_contributor_or_public
+
+logger = logging.getLogger(__name__)
+
+import xml.etree.ElementTree as ET
+
+@must_be_valid_project
+@must_have_addon('dryad', 'node')
+def check_dryad_doi(node_addon, **kwargs):
+    doi = request.args["doi"]
+    try:
+        d = Dryad_DataOne()
+        d.metadata(doi)
+    except HTTPError:
+        return {'result': False}
+    return {'result': True}
+
+
+@must_be_valid_project
+@must_have_addon('dryad', 'node')
+def set_dryad_doi(node_addon, **kwargs):
+    doi = request.args["doi"]
+    pid = kwargs['pid']
+    auth = kwargs['auth']
+    d = Dryad_DataOne()
+    try:
+        m = d.metadata(doi)
+    except HTTPError:
+        return redirect("/project/{}".format(pid))
```

Comment:
I'm a little confused by this redirect, but think a better understanding of the UI/feature set will help